etc/ci/performance/runner.py

```diff
@@ -0,0 +1,260 @@
+#!/usr/bin/env python3
+
+import argparse
+import itertools
+import json
+import os
+import subprocess
+from statistics import median, StatisticsError
+
+
+def load_manifest(filename):
+    with open(filename, 'r') as f:
+        text = f.read()
+    return list(parse_manifest(text))
+
+
+def parse_manifest(text):
+    return filter(lambda x: x != "" and not x.startswith("#"),
+                  map(lambda x: x.strip(), text.splitlines()))
+
+
+def execute_test(url, command, timeout):
+    print("Running test:")
+    print(command)
+    print("Timeout:{}".format(timeout))
+    try:
+        return subprocess.check_output(command, stderr=subprocess.STDOUT,
+                                       shell=True, timeout=timeout)
+    except subprocess.CalledProcessError as e:
+        print("Unexpected Fail:")
+        print(e)
+        print("You may want to re-run the test manually:\n{}".format(command))
+    except subprocess.TimeoutExpired:
+        print("Test timeout: {}".format(url))
+    return ""
+
+
+def get_servo_command(url, timeout):
+    ua_script_path = "{}/user-agent-js".format(os.getcwd())
+    test_cmd = ("timeout {timeout}s ./servo/servo '{url}'"
+                " --userscripts {ua} -x -o {png}").format(timeout=timeout,
+                                                          url=url,
+                                                          ua=ua_script_path,
+                                                          png="output.png")
+    return test_cmd
+
+
+def get_gecko_command(url, timeout):
+    test_cmd = ("timeout {timeout}s ./firefox/firefox/firefox"
+                " --display=:0 --no-remote"
+                " -profile ./firefox/servo {url}").format(timeout=timeout,
+                                                          url=url)
+    return test_cmd
+
+
+def parse_log(log, testcase=None):
+    blocks = []
+    block = []
+    copy = False
+    for line_bytes in log.splitlines():
+        line = line_bytes.decode()
+
+        if line.strip() == ("[PERF] perf block start"):
+            copy = True
+        elif line.strip() == ("[PERF] perf block end"):
+            copy = False
+            blocks.append(block)
+            block = []
+        elif copy:
+            block.append(line)
+
+    placeholder = {
```

Comment:
I think there used to be a comment explaining all the `-1`s, can you please put it back?